suite2p/classification/classify.py

```diff
@@ -1,53 +1,50 @@
-import os
 import numpy as np
 from pathlib import Path
 from .classifier import Classifier
 
 
-def classify(ops, stat, classfile=None, keys=['npix_norm', 'compact', 'skew']):
+def classify(save_path, stat, classfile=None, keys=None):
     """
     Applies classifier and saves output to iscell.npy. Also, saves stat.npy.
 
     Parameters
     ----------
-    ops : dictionary
-        'save_path'
-        (optional 'preclassify')
+    save_path : destination of output files
 
     stat : array of dicts
 
     classfile: string (optional)    
         path to classifier
 
+    keys: features (optional)
+        features
+
     Returns
     -------
 
-    ops : dictionary
-
-    stat : array of dicts
-
     iscell : array of classifier output
 
     """
+    if keys is None:
+        keys = []
+    keys = keys + ['npix_norm', 'compact', 'skew']
```

Comment:
this is wrong, you might need to use fewer keys --- why can't we use my implementation of this?

Reply:
Yep, you're right. I didn't consider the option of having fewer keys. 

Having `keys`' default parameter as a mutable is dangerous as shown in this [link](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/). For instance, the following code block fails: 

```
from suite2p import classification
import numpy as np
import copy
# stat with npix_norm, skew, compact
stat_file = np.load('/Users/chriski/Desktop/suite2p_ws/suite2p/plane0/stat.npy', allow_pickle=True)

# remove npix_norm
stat_no_npix = copy.deepcopy(stat_file)
for k in stat_no_npix:
    k.pop('npix_norm', None)

# remove skew from stat
stat_no_skew = copy.deepcopy(stat_file)
for ns in stat_no_skew:
    ns.pop('skew', None)

stat_no_compact = copy.deepcopy(stat_file)
for nc in stat_no_compact:
    nc.pop('compact', None)

classification.classify(save_path='/Users/chriski/Desktop/suite2p_ws', stat=stat_no_npix)
classification.classify(save_path='/Users/chriski/Desktop/suite2p_ws', stat=stat_no_skew)
classification.classify(save_path='/Users/chriski/Desktop/suite2p_ws', stat=stat_no_compact)

# just to check that stat_file has the default keys
print("stat_file has npix_norm value: {}".format(stat_file[0].get('npix_norm')))
print("stat_file has skew value: {}".format(stat_file[0].get('skew')))
print("stat_file has compact value: {}".format(stat_file[0].get('compact')))
classification.classify(save_path='/Users/chriski/Desktop/suite2p_ws', stat=stat_file)

```

The `remove` in classify.py ends up making the default `keys` list completely empty.

Reply:
Maybe it might be better to have default `keys` be a tuple and then convert it to a list within the classify func so that the keys could be removed safely??